js/js_test.go

```diff
@@ -597,3 +597,26 @@ func TestTypeSwitchJSObject(t *testing.T) {
 		}
 	}
 }
+
+type StructWithNonIdentifierJsTags struct {
+	object *js.Object
+	Name   string `js:"my name"`
+}
+
+func TestStructWithNonIdentifierJsTags(t *testing.T) {
+	s := StructWithNonIdentifierJsTags{
+		object: js.Global.Get("Object").New(),
+	}
+
+	const want = "Paul"
+
+	// externalise a value
+	s.Name = want
+
+	// internalise again
+	got := s.Name
+
+	if want != got {
+		t.Errorf("Value via non-identifier js tag field gave %q, want %q", got, want)
+	}
```

Comment:
Should we also add coverage for `js.Object.Get` and `js.Object.Set` methods? Are they equally affected?

Reply:
No, they are not affected: https://gopherjs.github.io/playground/#/osNIk-Yqbn

[Here's the relevant line in the compiler](https://github.com/gopherjs/gopherjs/blob/82b322028c96512b15077093b16a5f1c7ea897ac/compiler/expressions.go#L601-L610)